### PR TITLE
refactor(robot-server, api): robot server port /robot endpoints fast api

### LINKED ISSUE #5191: api: port robot endpoints from aiohttp to fastapi
- [ ] Port business logic to use fastapi endpoints in `robot-server`
- [ ] Port `api` tests to `robot-server` using starlette's testing client

### PR DESCRIPTION
**Pull request by amitlissack**
## overview

Implements these fine endpoints in robot-server fastapi
```
GET /robot/positions
POST /identify
POST /robot/move
POST /robot/home
GET /robot/lights
POST /robot/lights
```

## changelog

- api: move ThreadedAsyncLock out of the `server` package and into `hardware_control`
- api: add test for mount with z too low to aiohttp
- robot-server: move `/camera` to it's own router. still not implemented.
- robot-server: implement endpoints with tests.
- robot-server: tests are copied from api but split up into smaller bits. 

## review requests
- Is all the validation that was moved from `server.endpoints.control._validate_move_data` (now in model) correct? worth keeping?

## risk assessment

Low risk. Almost all the changes are in robot-server behind a feature flag.

closes #5191

### CODE REVIEW

File: robot-server/robot_server/service/models/control.py [hunk 1]
```diff
@@ -74,6 +77,18 @@ class RobotMoveTarget(BaseModel):
                           "to the specified mount. Required "
                           "if target is pipette")
 
+    @root_validator(pre=True)
+    def root_validator(cls, values):
+        points = values.get("point", [])
+        target = values.get("target")
+        if target == MotionTarget.mount and \
+                len(points) == 3 and points[2] < 30:
+            raise ValueError("Sending a mount to a z position lower than 30 "
```
Comment (sfoster1) [inline robot-server/robot_server/service/models/control.py]: 😬

Comment (sfoster1) [reply robot-server/robot_server/service/models/control.py]: (no change requested just 😬 )

File: robot-server/robot_server/service/routers/control.py [hunk 2]
```diff
@@ -12,62 +18,117 @@
...
-async def post_move_robot(robot_move_target: control.RobotMoveTarget)\
+async def post_move_robot(
+        robot_move_target: control.RobotMoveTarget,
+        hardware: HardwareAPILike = Depends(get_hardware),
+        motion_lock: ThreadedAsyncLock = Depends(get_motion_lock))\
         -> V1BasicResponse:
-    raise HTTPException(HTTPStatus.NOT_IMPLEMENTED, "not implemented")
+    """Move the robot"""
+    async with motion_lock:
+        await hardware.cache_instruments()  # type: ignore
```
Comment (sfoster1) [inline robot-server/robot_server/service/routers/control.py]: what does the `type: ignore` suppress here?

Comment (amitlissack) [reply robot-server/robot_server/service/routers/control.py]: I agree that it's gross, but I don't know what else to do really.

The issue is that I don't know how to type hardware in any meaningful way. `HardwareApiLike` ? `ThreadManager`? The linter yells that `cache_instruments` is not a member of `HardwareApiLike`, which is true. But it's not a member of `ThreadManager` either.

Comment (amitlissack) [reply robot-server/robot_server/service/routers/control.py]: @sfoster1 is that a sufficient answer?

File: robot-server/robot_server/service/routers/control.py [hunk 3]
```diff
@@ -12,62 +18,117 @@
...
 
 @router.get("/robot/lights",
             description="Get the current status of the OT-2's rail lights",
             response_model=control.RobotLightState)
-async def get_robot_light_state() -> control.RobotLightState:
-    raise HTTPException(HTTPStatus.NOT_IMPLEMENTED, "not implemented")
+async def get_robot_light_state(
+        hardware: HardwareAPILike = Depends(get_hardware)) \
+        -> control.RobotLightState:
+    on = hardware.get_lights()  # type: ignore
```
Comment (iansolano) [inline robot-server/robot_server/service/routers/control.py]: `robot_light_state` might be better name?

Comment (amitlissack) [reply robot-server/robot_server/service/routers/control.py]: A better name for what? `get_lights`?

Comment (iansolano) [reply robot-server/robot_server/service/routers/control.py]: instead of `on`

Comment (amitlissack) [reply robot-server/robot_server/service/routers/control.py]: Ohhh. Sorry.

File: robot-server/tests/service/routers/test_control.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    """Fixture for home handlers"""
+    async def mock_func(*args, **kwargs):
+        pass
+
+    hardware.home.side_effect = mock_func
+    hardware.home_plunger.side_effect = mock_func
+    return hardware
+
+
+def test_home_pipette(api_client, hardware_home):
```
Comment (iansolano) [inline robot-server/tests/service/routers/test_control.py]: not totally necessary but seems nice to test `left` and `right` just to keep it comprehensive no? almost serves as documentation in some way

**Pull request merged**